vcd/resource_vcd_org_user.go

```diff
@@ -0,0 +1,351 @@
+package vcd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func resourceVcdOrgUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdOrgUserCreate,
+		Read:   resourceVcdOrgUserRead,
+		Delete: resourceVcdOrgUserDelete,
+		Update: resourceVcdOrgUserUpdate,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdOrgUserImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCase("lower"),
+				Description:  "User's name. Only lowercase letters allowed. Cannot be changed after creation",
+			},
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Organization this user belongs to",
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    false,
+				Description: "Role within the organization",
+			},
+			"password": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+
+				ConflictsWith: []string{"password_file"},
+				Description: "The user's password. This value is never returned on read. " +
+					"It is inspected on create and modify. " +
+					"On modify, the absence of this element indicates that the password should not be changed.",
+			},
+			"password_file": &schema.Schema{
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      false,
+				ConflictsWith: []string{"password"},
+				Description:   "Name of a file containing the user's password.",
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's description",
+			},
+			"provider_type": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     govcd.OrgUserProviderIntegrated,
+				ForceNew:    false,
+				Description: "Identity provider type for this this user. One of: 'INTEGRATED', 'SAML', 'OAUTH'",
```

Comment:
Worth noting default value?

Reply:
The `Default` line above the description is not enough?

Reply:
Probably you are right. In the end the user simply won't see this description if he/she skips this field.